ui/src/app/utils/runtime-utils.tsx

```diff
@@ -265,29 +267,42 @@ export const useRuntimeStatus = (currentWorkspaceNamespace): [
 
   useEffect(() => {
     // Additional status changes can be put here
-    if (!!runtimeStatus) {
-      switchCase(runtimeStatus,
-        [RuntimeStatusRequest.Delete, async() => {
-          try {
-            await LeoRuntimeInitializer.initialize({workspaceNamespace: currentWorkspaceNamespace, maxCreateCount: 0});
-          } catch (e) {
-            // ExceededActionCountError is expected, as we exceed our create limit of 0.
-            if (!(e instanceof ExceededActionCountError ||
-                  e instanceof LeoRuntimeInitializationAbortedError)) {
-              throw e;
-            }
+    const targetStatus = switchCase(runtimeStatus,
+        [RuntimeStatusRequest.Delete, () => RuntimeStatus.Deleted],
```

Comment:
The goal in the deleted case should technically be `runtime === null` OR status is Deleted